newrelic/hooks/external_httpx.py

```diff
@@ -12,23 +12,56 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from newrelic.common.object_wrapper import wrap_function_wrapper
 from newrelic.api.external_trace import ExternalTrace
+from newrelic.api.transaction import current_transaction
+from newrelic.common.object_wrapper import wrap_function_wrapper
 
 
 def bind_request(request, *args, **kwargs):
     return request
 
 
 def sync_send_wrapper(wrapped, instance, args, kwargs):
+    transaction = current_transaction()
     request = bind_request(*args, **kwargs)
-    with ExternalTrace('httpx', str(request.url), request.method):
+    connection = instance
+
+    with ExternalTrace("httpx", str(request.url), request.method) as tracer:
+        # Add the tracer to the connection object. The tracer will be
+        # used in getresponse() to add back into the external trace,
+        # after the trace has already completed, details from the
+        # response headers.
+        if hasattr(tracer, "generate_request_headers"):
+            outgoing_headers = tracer.generate_request_headers(transaction)
+            for header_name, header_value in outgoing_headers:
+                # User headers should override our CAT headers
+                if not header_name in request.headers:
+                    request.headers[header_name] = header_value
+
+            connection._nr_external_tracer = tracer
```

Comment:
No need to store the tracer here, the response will be processed directly on line 43 below and won't need any type of "restore" logic.
```suggestion
```